src/ethereum/london/vm/instructions/system.py

```diff
@@ -110,17 +116,15 @@ def generic_create(
         accessed_storage_keys=evm.accessed_storage_keys.copy(),
     )
     child_evm = process_create_message(child_message, evm.env)
-    evm.children.append(child_evm)
+
     if child_evm.has_erred:
-        push(evm.stack, U256(0))
+        incorporate_child_evm_error(evm, child_evm)
         evm.return_data = child_evm.output
```

Comment:
Should the `incorporate_child_*` functions handle the `return_data`?

Reply:
The behaviour for `return_data` for `CREATE*` vs `CALL*` opcodes, so has to be handled separately.